docs/remote-access/wireguard.md

```diff
@@ -11,4 +11,112 @@ This page will cover two WireGuard architectures. The first is running WireGuard
 
 ## WireGuard Server on a VPS
 
-* Todo
\ No newline at end of file
+This section will talk about setting up a VPS and then running a client on your network to setup the connection to the VPS and allow the vpn to work without a firewall opening. 
+
+The first step is to decide on a virtual host provider and create a server. When you decide on a provider and create a host. Install Wireguard using the instructions provided on the [Wireguards Documentation](https://www.wireguard.com/install/)
+
+### Server Configuration Setup:
+First create a folder to contain all the details of our configuration.
+```shell
+mkdir wireguard
+cd wireguard
+```
+
+Now you must create the keys you have to options on how many keys you need to make. The simplest allows for just two keys but wireguard also allows you to provide a preshared key as well. For this configuration we will work with the preshared key but if you do not want to you can remove those files from the configuration.
+
+```shell
+wg genkey | tee cloud_key.priv | wg pubkey > cloud_key.pub
+wg genkey | tee client_1_key.priv | wg pubkey > client_1_key.pub
+wg genpsk > client_1_cloud_preshared_key.psk
+```
+
+```shell
+vim /etc/wireguard/wg0.conf
+```
+
+When you create the file you will copy the following configuration:
+```
+[Interface]
+# Cloud
+Address = 192.168.1.1/24
+ListenPort = 51902
+PrivateKey = << Private Key - cloud_key.priv >>
+
+PostUp = iptables -A FORWARD -i %i -j ACCEPT ; iptables -A FORWARD -o %i -j ACCEPT; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+PostDown = iptables -D FORWARD -i %i -j ACCEPT ; iptables -D FORWARD -o %i -j ACCEPT; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE
+
+[Peer]
+# Client 1
+PublicKey = << Public Key - client_1_key.pub >>
+PresharedKey = << Preshared Key - client_1_cloud_preshared_key.psk >>
+AllowedIPs = 192.168.1.2/32,10.0.0.0/16
+Endpoint = << Client 1 DDNS >>:51902
+
+[Peer]
+# Client 2
+PublicKey = << Public Key - client_2_key.pub >>
+PresharedKey = << Preshared Key - client_2_cloud_preshared_key.psk >>
+AllowedIPs = 192.168.1.3/32
+Endpoint = << Client 2 DDNS >>:51902
+
+```
+
+Lets talk about the information provided above. First look at the Interface configuration. The first item we have a comment to provide a name for the host we are configuring. This makes it a little easier to identify the configuration. The next line is the IP address for the wireguard network. When choosing the subnet you should keep this separate from you LAN subnet. When we setup the service we need to tell it what port to bind to, in the configuration above we use 51902. Finally the last item is the PrivateKey that the server will use, which we created above. 
+
+The last two items in the Interface definition are the commands that will be run after the interface is stood up or taken down. The commands included in the provided configuration allows anything put over the wireguard network to this interface to be forwarded to the network it is on.
```

Comment:
The last sentence of the first paragraph states "Finally the last item" but then the next paragraph starts with "The last two items", I feel like this is a bit confusing as you now have two sets of last items. Perhaps it would be better to say:
```
Next is the PrivateKey that the server will use, which we created above. 
```